service/payment.go

```diff
@@ -121,45 +122,90 @@ func (service *PaymentService) GetPaymentSession(w http.ResponseWriter, req *htt
 		return
 	}
 
-	paymentResource, err := service.DAO.GetPaymentResource(id)
-	if paymentResource == nil {
-		log.Info(fmt.Sprintf("payment session not found. id: %s", id))
-		w.WriteHeader(http.StatusForbidden)
+	paymentSession, httpStatus, err := (*PaymentService).getPaymentSession(service, id)
+	if err != nil {
+		w.WriteHeader(httpStatus)
+		log.ErrorR(req, err)
 		return
 	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	err = json.NewEncoder(w).Encode(paymentSession)
 	if err != nil {
-		log.ErrorR(req, fmt.Errorf("error getting payment resource from db: [%v]", err))
-		w.WriteHeader(http.StatusInternalServerError)
+		log.ErrorR(req, fmt.Errorf("error writing response: %v", err))
 		return
 	}
 
-	costs, err := getCosts(w, req, paymentResource.Data.Links.Resource, &service.Config)
+	log.InfoR(req, "Successfully GET request for payment resource: "+id)
```

Comment:
As mentioned above could you use log.Data{"payment_id": paymentResource.ID} please.